src/groovy/org/pih/warehouse/api/PutawayItem.groovy

```diff
@@ -40,6 +41,18 @@ class PutawayItem {
         return currentBins
     }
 
+    String getPreferredBin() {
+        String preferredBin = ""
+        if (availableItems) {
+            locations.each { location ->
+                preferredBin = availableItems?.find { it?.binLocation?.id == location?.id }?.collect {
+                    it?.binLocation?.name
+                }?.join("")
```

Comment:
You should return the first match, check the ticket description

Reply:
I think it should look like this:
`String preferredBin = availableItems?.find { availableItem -> locations?.find { it.id == availableItem?.binLocation?.id } }?.binLocation?.name
return preferredBin || ""`